### PR TITLE
[BE] Propose candidates by technology to invite

### PR DESCRIPTION
**Pull request by nspc97**
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added route /user/list filtered by technologies & sorted by seniority

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -396,4 +398,81 @@ public function userChangePassword(
         $userRepository->save($userChangePassword);
         return new JsonResponse(['message' => 'Password successfully changed!'], Response::HTTP_OK);
     }
+
+    /**
+     * Change password of User.
```
Comment (MihaiDruta) [inline src/Controller/UserController.php]: Wrong description...

**Review by MihaiDruta — APPROVED** [review]

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -396,4 +398,81 @@ public function userChangePassword(
...
+        UserListFilter $userListFilter,
+        UserListSort $userListSort,
+        UserRepository $userRepository
+    ): JsonResponse {
+
+        $filter = $request->query->get('filter');
+        $userListFilter->setFilterFields((array)$filter);
+
+        $sorting = $request->query->get('sortBy');
+        $userListSort->setSortingFields((array)$sorting);
```
Comment (SpartakusMd) [inline src/Controller/UserController.php:469]: Pagination is missing. In order to have the possibility to extract all the items in one page we should handle the case when `per_page=-1`. If it is exactly `-1` => return all the items.

File: src/Repository/UserRepository.php [hunk 3]
```diff
@@ -68,4 +70,27 @@ public function getApplicantsForActivity(ApplicantsListSort $applicantsListSort,
...
+     * @return QueryBuilder
+     */
+    public function getUserList(
+        UserListFilter $userListFilter,
+        UserListSort $userListSort
+    ): QueryBuilder {
+        $queryBuilder = $this->createQueryBuilder('user');
+        $queryBuilder
+            ->select('user');
+        if ($userListFilter->technology !== null) {
```
Comment (SpartakusMd) [inline src/Repository/UserRepository.php]: Use `if (!empty($userListFilter->technology))`. If someone will do a request `...?filter[technology]=&...` (note the missing value for technology) you will get a database error `unexpected closing parenthesis in "technology IN ()"`

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Handlers/UserHandler.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        $this->userRepository = $userRepository;
+    }
+
+    public function getUserListPaginated(
+        UserListPagination $userListPagination,
+        UserListSort $userListSort,
+        UserListFilter $userListFilter
+    ): array {
+
+        if ($userListPagination->pageSize === -1) {
```
Comment (SpartakusMd) [inline src/Handlers/UserHandler.php]: This check should be done in `getPaginatedUserList()`.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Pull request merged**